### ISSUE TITLE
Controller disconnects as soon as it's plugged in

### ISSUE THREAD

**Issue by Kommynct** [YARC-Official/YARG#773]
My controller isn't detected, if I unplug it and plug it back in, I get the notification that it's disconnected and connected at the same time.

![image](https://github.com/YARC-Official/YARG/assets/13646790/a709d94d-f56b-4895-bfb8-3cfba8f68b2a)

**Comment by NicKoehler** [comment]
I'm on Arch Linux too, but the game is from the official AppImage launcher. I still have the same issue, but with another guitar.

![image](https://github.com/YARC-Official/YARG/assets/53040044/c5d649a8-3434-48d7-a142-2dc74db26b8f)

**Comment by EliteAsian123** [comment]
Did y'all follow the additional setup instructions needed for Linux in the readme?

**Comment by NicKoehler** [comment]
Yes.

- [x] Installed additional packages with `pacman -S hidapi systemd-libs`
- [X] Added the udev rule `KERNEL=="hidraw*", TAG+="uaccess"`

If there is something I can do to debug this issue, I'm willing to help!

**Comment by NicKoehler** [comment]
Clone hero:
- [x] PS2 GUITAR (connected using an usb adapter)
- [x] XBOX GUITAR (connected directly to the usb port)

Yarg:
- [x] PS2 GUITAR (connected using an usb adapter)
- [ ] XBOX GUITAR (connected directly to the usb port)

**Comment by TheNathannator** [comment]
Xbox stuff isn't reliably supported currently on Linux, there's some hacks going on to disable the native Unity controller support and that's the "instant disconnect" you're seeing. I tried to make sure Xbox controllers would make it through, but unfortunately that doesn't work on all distros.

**Comment by Kommynct** [comment]
> Did y'all follow the additional setup instructions needed for Linux in the readme?

It wouldn't have worked in clone hero if I hadn't.

My controller isn't an xbox controller, i'd like an option to enable the native unity controller support because again, this works absolutely flawlessly in clone hero.

**Comment by EliteAsian123** [comment]
As far as I know, Clone Hero uses Rewired and not the native Unity controller support, so unfortunately that wouldn't help with this problem.

**Comment by TheNathannator** [comment]
> It wouldn't have worked in clone hero if I hadn't.

Yes it would have, CH doesn't use the same input system we use. We use the newer Unity input system and handle bindings more manually, while CH uses Rewired which is much more generic.

For now, if you want to re-enable the native Unity input support, uninstall `hidapi`, that'll make our custom backend not initialize. Just beware that it *is not* perfect, and will likely miss certain controls. The custom HID backend exists to fix these issues, there just isn't an equivalent for evdev on Linux yet.

**Comment by NicKoehler** [comment]
hidapi is required for a lot of other packages on my system. I tried to remove it temporarily to see if my guitar would be detected, but as you said, certain controls do not work, like the red button and the strums.

**Comment by AlenaEK** [comment]
I had this problem too. My solution was to just install the game on Windows instead lol.

**Comment by RaysandraRexxia** [comment]
I'm having this problem on Windows 10 with the RB4 Xbox One controller
